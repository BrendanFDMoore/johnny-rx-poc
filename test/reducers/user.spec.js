import { REHYDRATE } from 'redux-persist/constants';

import reducers from 'reducers';
import { ActionTypes } from 'constants/index';

describe('User', () => {
  it('should return the initial state', () => {
    expect(reducers.user(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${REHYDRATE} with payload`, () => {
    expect(reducers.user(undefined, { type: REHYDRATE, payload: { user: { logged: true } } })).toMatchSnapshot();
  });

  it(`should handle ${REHYDRATE} without payload`, () => {
    expect(reducers.user(undefined, { type: REHYDRATE, payload: { } })).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.USER_LOGIN_SUCCESS}`, () => {
    expect(reducers.user(undefined, { type: ActionTypes.USER_LOGIN_SUCCESS })).toMatchSnapshot();
  });

  it(`should handle ${ActionTypes.USER_LOGOUT_SUCCESS}`, () => {
    expect(reducers.user(undefined, { type: ActionTypes.USER_LOGOUT_SUCCESS })).toMatchSnapshot();
  });
});
