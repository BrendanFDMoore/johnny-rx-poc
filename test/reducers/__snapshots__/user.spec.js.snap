exports[`User should handle USER_LOGIN_SUCCESS 1`] = `
Object {
  "logged": true,
  "rehydrated": false,
}
`;

exports[`User should handle USER_LOGOUT_SUCCESS 1`] = `
Object {
  "logged": false,
  "rehydrated": false,
}
`;

exports[`User should handle persist/REHYDRATE with payload 1`] = `
Object {
  "logged": true,
  "rehydrated": true,
}
`;

exports[`User should handle persist/REHYDRATE without payload 1`] = `
Object {
  "logged": false,
  "rehydrated": true,
}
`;

exports[`User should return the initial state 1`] = `
Object {
  "logged": false,
  "rehydrated": false,
}
`;
